holoviews/core/data/interface.py

```diff
@@ -50,22 +50,31 @@ def initialize(cls, eltype, data, kdims, vdims, datatype=None):
             kdims = pvals.get('kdims') if kdims is None else kdims
             vdims = pvals.get('vdims') if vdims is None else vdims
 
+        if datatype is None:
+            datatype = eltype.datatype
+
         # Process Element data
         if isinstance(data, NdElement):
             kdims = [kdim for kdim in kdims if kdim != 'Index']
-        elif hasattr(data, 'interface') and issubclass(data.interface, Interface):
-            data = data.data
+        elif (hasattr(data, 'interface') and issubclass(data.interface, Interface)):
```

Comment:
Seems a bit odd - I guess this is because not all HoloViews elements use interfaces (e.g annotations?)